src/app/pages/questions/components/question/audio-input/audio-input.component.ts

```diff
@@ -69,10 +72,15 @@ export class AudioInputComponent implements OnDestroy, OnInit {
     if (!this.isRecording()) {
       this.recordAttempts++
       if (this.recordAttempts <= DefaultMaxAudioAttemptsAllowed) {
-        this.startRecording().catch(e => this.showTaskInterruptedAlert())
+        this.startRecording()
+          .then(() =>
+            this.usage.sendGeneralEvent(UsageEventType.RECORDING_STARTED)
+          )
+          .catch(e => this.showTaskInterruptedAlert())
       }
     } else {
       this.stopRecording()
+      this.usage.sendGeneralEvent(UsageEventType.RECORDING_STOPPED)
```

Comment:
I think this is to do with the usage service but why are we ignoring the promise returned by analytics service in UsageService -> sendGeneralEvent()? It could be useful if the usage service also returned a promise so we can chain events and also log errors in case the promise failed?

Reply:
I didn't chain it because I didn't want it to block the other functions/processes if it was unable to send events? 
The firebase analytics logEvent() already logs the result of the promise, and I have added the logging for sending the event to Kafka.
What do you think?

Reply:
Ok sounds good.